packages/mui-base/src/SliderUnstyled/SliderUnstyled.tsx

```diff
@@ -281,8 +282,8 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled<
             })}
             style={{
               ...style,
-              pointerEvents: disableSwap && active !== index ? 'none' : undefined,
-              ...thumbProps.style,
+              // @ts-ignore
+              ...thumbProps.style(index),
```

Comment:
It doesn't make sense. `style` property is not callable.

A better way would be to get the `index` in the `useSlider` using the `data-index` attribute by querying the thumb node with the `thumb` class.

Reply:
Yeah it is resolved, I've moved to styling computation to individual components from useSlider hook as I felt it difficult to get index without passing from component.

You can review now, didn't added tests as I'm not sure how to test